src/screens/tabs/NFTs/screens/NftDetail/index.tsx

```diff
@@ -0,0 +1,166 @@
+import { NFTDetails } from '@psychedelic/dab-js';
+import { StackHeaderProps } from '@react-navigation/stack';
+import { t } from 'i18next';
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
+import { Linking, Platform, View } from 'react-native';
+import { ScrollView } from 'react-native-gesture-handler';
+import { Modalize } from 'react-native-modalize';
+
+import Button from '@/components/buttons/Button';
+import RainbowButton from '@/components/buttons/RainbowButton';
+import { ActionSheet, Badge, NftDisplayer, Text } from '@/components/common';
+import ModalHeader from '@/components/navigation/ModalHeader';
+import { ICNS_CANISTER_ID } from '@/constants/canister';
+import { FontStyles } from '@/constants/theme';
+import useGetType from '@/hooks/useGetType';
+import DownloadIcon from '@/icons/material/Download.svg';
+import ViewIcon from '@/icons/material/View.svg';
+import { ModalScreenProps } from '@/interfaces/navigation';
+import Routes from '@/navigation/Routes';
+import { useAppSelector } from '@/redux/hooks';
+import { getNFTDetails } from '@/services/DAB';
+import { downloadFile } from '@/utils/filesystem';
+import { deleteWhiteSpaces } from '@/utils/strings';
+
+import Section from './components/Section';
+import styles from './styles';
+
+const header = (props: StackHeaderProps, showBack?: boolean) => (
+  <ModalHeader {...props} showBack={showBack} />
+);
+
+function NftDetail({ route, navigation }: ModalScreenProps<Routes.NFT_DETAIL>) {
+  const { canisterId, index, showBack } = route.params;
+  const { collections } = useAppSelector(state => state.user);
+  const collection = collections.find(c => c.canisterId === canisterId);
+  const selectedNFT = collection?.tokens?.find(token => token.index === index);
+  const isICNS = selectedNFT?.canister === ICNS_CANISTER_ID;
+  const nftName = `${collection?.name || ''} #${index}`;
+  const type = useGetType(selectedNFT?.url);
+
+  const actionSheetRef = useRef<Modalize>(null);
+  const [isDownloading, setIsDownloading] = useState(false);
+  const [nftDetails, setNFTDetails] = useState<NFTDetails<string | bigint>>();
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      title: isICNS ? selectedNFT?.name : `#${selectedNFT?.index}`,
+      header: (props: StackHeaderProps) => header(props, showBack),
+    });
+  }, [selectedNFT]);
+
+  useEffect(() => {
+    if (selectedNFT && !isICNS) {
+      getNFTDetails({
+        index,
+        canister: selectedNFT.canister,
+        standard: selectedNFT.standard,
+      })
+        .then(details => setNFTDetails(details))
+        .catch(() => {});
```

Comment:
is this ok?

Reply:
just to cover a crash in case the controller fails, happened before updating to the new version